### PR TITLE
Fix the horizontal communication between NATS nodes in a NATS cluster (production profile)

### PR DESCRIPTION
**Pull request by radufa**
**Description**

Changes proposed in this pull request:

- All NATS nodes in a production cluster should use the same cluster name
- The "podAntiAffinity" should work again

**Related issue(s)**

See also: 
https://github.com/kyma-project/kyma/issues/12866
https://github.com/kyma-project/kyma/issues/12825

### CODE REVIEW

File: resources/eventing/charts/nats/values.yaml [hunk 1]
```diff
@@ -163,13 +163,16 @@ podAnnotations: {
 statefulSetAnnotations: {}
 
 # Labels to add to the pods of the NATS StatefulSet
-statefulSetPodLabels: {}
+statefulSetPodLabels: {
+  nats_cluster: eventing-nats
+}
```
Comment (k15r) [inline resources/eventing/charts/nats/values.yaml:168]:
```suggestion
statefulSetPodLabels:
  nats_cluster: eventing-nats
```

File: resources/eventing/charts/nats/templates/statefulset.yaml [hunk 2]
```diff
@@ -33,7 +33,9 @@ spec:
       labels:
         {{- include "nats.selectorLabels" . | nindent 8 }}
         {{- if .Values.statefulSetPodLabels }}
-        {{ toYaml .Values.statefulSetPodLabels | indent 8 }}
+        {{- range $key, $value := .Values.statefulSetPodLabels }}
+        {{ $key }}: {{ $value | quote }}
+        {{- end }}
```
Comment (k15r) [inline resources/eventing/charts/nats/templates/statefulset.yaml:38]: is there no easier version to achieve the same, without manually quoting and iterating over the list of Labels?

File: resources/eventing/charts/nats/values.yaml (hunk 1 above)
Comment (radufa) [reply resources/eventing/charts/nats/values.yaml]: Done

File: resources/eventing/charts/nats/templates/statefulset.yaml (hunk 2 above)
Comment (radufa) [reply resources/eventing/charts/nats/templates/statefulset.yaml]: Done

**Review by k15r — APPROVED** [review]

**Pull request merged**